src/main/java/servlets/UploadReceiptServlet.java

```diff
@@ -53,8 +54,33 @@
   private static final String DEV_SERVER_BASE_URL = "http://0.0.0.0:80";
   // Matches JPEG image filenames.
   private static final Pattern validFilename = Pattern.compile("([^\\s]+(\\.(?i)(jpe?g))$)");
-  // Logs to System.err by default.
-  private static final Logger logger = Logger.getLogger(UploadReceiptServlet.class.getName());
+
+  private static Logger logger;
```

Comment:
Okay to keep one instance of logger (static) but I would also make this final.

Reply:
Marking the logger both final and static will prevent assigning it a value in the constructor since it is called for every instance. Should I only mark it as final then?

Reply:
I don't think it's necessary to create a new instance of the logger for each request so I would keep it static and final. I also think it's not super necessary to check the log message since you check the error status already. If you do want to check the log message though, there are options  to testing the logging by creating a mock appender.